### PR TITLE
Make restore warnings as errors.

### PR DESCRIPTION
**Pull request by codito**
Remove explicit reference to netcoreapp 1.0.0 package, instead use the latest
available in the sdk.

### CODE REVIEW

File: src/package/package/package.csproj [hunk 1]
```diff
@@ -5,9 +5,8 @@
   </PropertyGroup>
   <Import Project="$(TestPlatformRoot)scripts/build/TestPlatform.Settings.targets" />
   <PropertyGroup>
-    <TargetFrameworks>net46;netcoreapp1.0</TargetFrameworks>
+    <TargetFrameworks>netcoreapp1.0;net46</TargetFrameworks>
     <AssemblyName>package</AssemblyName>
-    <PackageTargetFallback Condition=" '$(TargetFramework)' == 'netcoreapp1.0' ">$(PackageTargetFallback);dnxcore50;netstandardapp1.0;portable-net45+win8;portable-net45+wp80+win8+wpa81+dnxcore50</PackageTargetFallback>
```
Comment (harshjain2) [inline src/package/package/package.csproj]: Is this not required now? I have seen it used at so many other csprojs as well.

File: test/datacollector.UnitTests/datacollector.UnitTests.csproj [hunk 2]
```diff
@@ -40,11 +40,11 @@
   </ItemGroup>
   <ItemGroup>
     <PackageReference Include="System.Xml.XmlDocument">
-      <Version>4.0.1</Version>
+      <Version>4.3.0</Version>
```
Comment (harshjain2) [inline test/datacollector.UnitTests/datacollector.UnitTests.csproj]: Just a bit curious, is this upgrade required or is it just done for upgrading to latest version of this package?

**Review by harshjain2 — APPROVED** [review]

File: test/datacollector.UnitTests/datacollector.UnitTests.csproj (hunk 2 above)
Comment (codito) [inline test/datacollector.UnitTests/datacollector.UnitTests.csproj]: This used to be an restore warning :)

File: src/package/package/package.csproj (hunk 1 above)
Comment (codito) [inline src/package/package/package.csproj]: Not required since this is a package project.

**Pull request merged**